crates/turbopack-css/src/chunk/writer.rs

```diff
@@ -1,108 +1,135 @@
                     imported_content.imports.iter().cloned().collect(),
                     (inner_indent, close),
                 ));
             }
             Some(CssImport::External(url_vc)) => {
-                external_imports.push(url_vc);
+                result.external_imports.push(url_vc);
             }
             None => {
-                let content = &*(*content_vc).await?;
-                writeln!(writer, "{}", content.inner_code)?;
-                writer.pop_indent(*indent)?;
-                writeln!(writer, "{}", close)?;
+                let mut code = CodeBuilderWithIndent::default();
+                let id = &*chunk_item.to_string().await?;
+                code.push_str(&format!("/* {} */\n", id), &indenter)?;
+
+                let content = chunk_item.content().await?;
+                code.push_source(&content.inner_code, content.source_map, &indenter)
+                    .await?;
+                indenter.pop(*indent);
+                code.push_str(&format!("\n{}\n", close), &indenter)?;
+
+                result.codes.push(code.build().cell());
                 stack.pop();
             }
         }
     }
 
-    Ok(external_imports)
+    Ok(result.cell())
 }
 
-pub struct WriterWithIndent<T: Write> {
-    writer: T,
-    indent_str: String,
-    needs_indent: bool,
+#[derive(Default)]
+struct CodeBuilderWithIndent {
+    code: CodeBuilder,
 }
 
-impl<T: Write> WriterWithIndent<T> {
-    pub fn new(buffer: T) -> Self {
-        Self {
-            writer: buffer,
-            indent_str: "".to_string(),
-            needs_indent: true,
-        }
-    }
-
-    pub fn push_indent(&mut self, indent: usize) -> std::fmt::Result {
-        self.indent_str += &" ".repeat(indent);
+#[derive(Default)]
+struct Indent(String);
 
-        Ok(())
+impl Indent {
+    pub fn push(&mut self, n: usize) {
+        self.0 += &" ".repeat(n);
     }
 
-    pub fn pop_indent(&mut self, indent: usize) -> std::fmt::Result {
-        self.indent_str = " ".repeat(self.indent_str.len().saturating_sub(indent));
+    pub fn pop(&mut self, n: usize) {
+        self.0 = " ".repeat(self.0.len().saturating_sub(n));
+    }
 
-        Ok(())
+    pub fn to_str(&self) -> &str {
+        &self.0
     }
 }
 
-impl<T: Write> Write for WriterWithIndent<T> {
-    #[inline]
-    fn write_str(&mut self, s: &str) -> std::fmt::Result {
-        for c in s.chars() {
-            self.write_char(c)?;
-        }
-
+impl CodeBuilderWithIndent {
+    pub async fn push_source(
+        &mut self,
+        code: &Rope,
+        map: Option<ParseResultSourceMapVc>,
+        indent: &Indent,
+    ) -> Result<()> {
+        let indented = self.wrap_indent(&code.to_str()?, indent)?.into();
+        let indented_map = if let Some(map) = map {
+            Some(
+                ParseResultSourceMap::with_indent(&*map.await?, indent.to_str().len() as u32)
+                    .cell()
+                    .as_generate_source_map(),
```

Comment:
```suggestion
                map.with_indent(indent.to_str().len() as u32).as_generate_source_map(),
```

So allow caching this need to be a method on the Vc

Reply:
The indention for source map also don't needed since we omit the indention now.